server/utils/applicationUtils.ts

```diff
@@ -31,6 +31,18 @@ export const submittedApplicationTableRows = (
   })
 }
 
+export const prisonDashboardTableRows = (applications: Array<Cas2ApplicationSummary>): Array<TableRow> => {
+  return applications.map(application => {
+    return [
+      nameAnchorElement(application.personName, application.id),
+      textValue(application.nomsNumber),
+      textValue(application.createdByUserName),
+      textValue(DateFormats.isoDateToUIDate(application?.hdcEligibilityDate, { format: 'medium' })),
```

Comment:
is `application` here always defined, so doesn't need the `?` 
`hdcEligibilityDate` looks like it is nullable though, do we need an example in the unit test of one without a date?

Reply:
(and if so, how should we handle that?)

Reply:
actually there may be some submitted applications without the date that were submitted before we introduced that field